### PR TITLE
fix: add component classes to modal

### PR DESCRIPTION
**Pull request by AnnaRybkina**
Styles are migrated from https://github.com/fabric-ds/css/blob/next/src/components/modal.css

Default|Justified title
---|---
image|image

Vue specific transition classes were migrated from https://github.com/fabric-ds/css/blob/next/src/components/modal.css#L29

### CODE REVIEW

File: components/modal/w-modal.vue [hunk 1]
```diff
@@ -1,30 +1,138 @@
...
-  <transition name="fade">
-    <div :class="c.backdrop" v-if="showModal" @click.self="emitDismiss" ref="backdropEl" data-test="backdrop">
-      <transition name="slide">
-        <div v-if="showContent" :class="c.modal" tabindex="-1" aria-modal="true" aria-labelledby="w-modal-title" role="dialog" ref="modalEl">
-          <div :class="[c.title, headerClasses]">
-            <transition-group name="w-modal-title">
+  <transition 
+    name="fade"
+    leave-active-class="transition-opacity-200 ease-in-out"
+    enter-active-class="transition-opacity-200 ease-in-out"
```
Comment (AnnaRybkina) [inline components/modal/w-modal.vue:97]: Vue specific transition classes, imho, should not be migrated to the component classes. Wdyt?

Comment (magnuh) [reply components/modal/w-modal.vue:97]: Why is there vue specific classes in the first place? 🤷 😄

Comment (AnnaRybkina) [reply components/modal/w-modal.vue:97]: 🤷 just migrating and applying styles here

Comment (AnnaRybkina) [reply components/modal/w-modal.vue:97]: It would be nice to align everything in the future. Like other stuff 😓

Comment (magnuh) [reply components/modal/w-modal.vue:97]: Will be odd to have different behaviour on "the same" component on a site, depending on which framework is used... So I think you are correct, leave it out of the component classes... unless it is the desired behaviour? 😄

**Review by Skadefryd — APPROVED** [review]

**Pull request merged**